### ISSUE TITLE
[BUG] CUT-Gizmo: wrong flap-tolerance-gap-width

### ISSUE THREAD

**Issue by discip** [SoftFever/OrcaSlicer#6878]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

latest 2.2.0-beta

### Operating System (OS)

Linux, Windows

### How to reproduce

![image](https://github.com/user-attachments/assets/f0182c34-672c-451c-805e-5e9ffe079721)

### Actual results

The gap width of the slanted part of the dovetail doesn't correspond to the set value.

### Expected results

Obviously the opposite of the actual result. 😊

### Project file & Debug log uploads

I tried to find out where I had to adjust the calculation, but unfortunately I couldn't understand how the wrong value came about.

What I measured:
At an angle of **`90°`** and a gap width of **`1 mm`** the gap on the flaps was only **`0.5 mm`** instead of **`1 mm`**.
At an angle of **`60°`** and a gap width of **`1 mm`** the gap on the flaps was only **`0.433 mm`** instead of **`1 mm`**.

I assumed this line was responsible for the mentioned calculation:
https://github.com/SoftFever/OrcaSlicer/blob/38cd758e948e5c2f9edddd5b281570b9e9cc0a8b/src/slic3r/GUI/Gizmos/GLGizmoCut.cpp#L728

@Noisyfox
Maybe you have an idea about the root cause?

**Comment by Noisyfox** [comment]
Does this happen on 2.1.1 as well?

**Comment by discip** [comment]
> Does this happen on 2.1.1 as well?

Yes, unfortunately exactly the same behavior.

**Comment by discip** [comment]
> Does this happen on 2.1.1 as well?

Yes, unfortunately exactly the same behavior.

**`BambuStudio`** and **`PrusaSlicer`** also have this issue.

**Comment by Noisyfox** [comment]
I think the math works correctly here. The width tolerance here is not how much the two side faces are apart from each other perpendicular:
![image](https://github.com/user-attachments/assets/92f232ec-1ae2-443c-933d-3c0d494faf1e)

but rather how far apart horizontally:
![image](https://github.com/user-attachments/assets/f730b52e-31a9-456e-a65c-9daf5c33aa78)

And also, since the dovetail has two sides facing side ways, this tolerance is the amount with both sides addes up.

**Comment by discip** [comment]
> And also, since the dovetail has two sides facing side ways, this tolerance is the amount with both sides adds up.

Yeah, that makes absolutely sense if you consider that between the other 2 horizontal faces there is no gap at all.

But that (Gap only on one set of opposing faces.) makes this approach of creating a groove somewhat fishy.

The better solution would be a gap that has a homogeneous distance between all opposing faces like the following, except without the radii (These are only to make clear that the distance between the slanted faces should be measured perpendicular to the faces).

![image](https://github.com/user-attachments/assets/156c75a5-243c-49a3-b2ff-da89b72579b1)

I hope I was able to make myself clear.

thanks in advance

**Comment by Noisyfox** [comment]
However, consider how FDM tend to be more inaccurate on XY direction than Z, it's somewhat make sense to calculate the tolerance on the xy direction.

**Comment by discip** [comment]
Ah, that's a good point.

I was having a case in mind where the cut would be placed on the XZ or YZ plane so that the profile would lie on the XY plane and in this direction not Z direction would be involved.

Do you think it would make sense to differentiate between those cases?
I'm aware of the fact, that there would most likely be scenarios where someone would place a cut diagonally instead of vertically or horizontally. 😊

Would it be worth the effort?

**Comment by Noisyfox** [comment]
Ah how stupid I am! I forgot that the cut plane can be tilted... Then my theory about the xy inaccuracy no longer holds. I think it might be better to do what you suggested, though this could be a breaking change and some might find their cut grooves no longer fit.

**Comment by discip** [comment]
Is there a possible solution for at least the two cases I mentioned?

**Comment by discip** [comment]
Is there a possible solution for at least the two cases I mentioned? 😅
